File: play/scala/rest-api/app/controllers/HomeController.scala
package controllers

import java.util.concurrent.atomic.AtomicLong
import javax.inject._

import play.api.mvc._
import play.api.libs.json._

/**
 * This controller creates an `Action` to handle HTTP requests to the
 * application's home page.
 */
@Singleton
class HomeController @Inject()(cc: ControllerComponents) extends AbstractController(cc) {

  val counter: AtomicLong = new AtomicLong()

  /**
   * Create an Action to render an HTML page.
   *
   * The configuration in the `routes` file means that this method
   * will be called when the application receives a `GET` request with
   * a path of `/`.
   */
  case class Greeting(count: Long, greeting: String)
  implicit val greetingWrites = Json.writes[Greeting]

  def index(guest: String) = Action {
    Ok(Json.toJson(Greeting(counter.incrementAndGet(), s"Hey, $guest")))
  }

}

File: play/java/rest-api/app/controllers/HomeController.java
package controllers;

import play.libs.Json;
import play.mvc.*;

import java.util.concurrent.atomic.AtomicLong;

/**
 * This controller contains an action to handle HTTP requests
 * to the application's home page.
 */
public class HomeController extends Controller {

    final AtomicLong counter = new AtomicLong();

    /**
     * An action that renders an HTML page with a welcome message.
     * The configuration in the <code>routes</code> file means that
     * this method will be called when the application receives a
     * <code>GET</code> request with a path of <code>/</code>.
     */

    public Result greeting(String guest) {

        return ok(Json.toJson(new Greeting(counter.incrementAndGet(), guest)));

    }




    public static class Greeting {
        public final Long count;
        public final String greeting;

        Greeting(Long count, String guest){
            this.count = count;
            this.greeting = String.format("Hello, %s!", guest);
        }
    }

}
